sdk/helper/awsutil/region.go

```diff
@@ -24,50 +26,47 @@ variable (and not `AWS_REGION`, while the golang SDK does _mostly_ the opposite
 reads the region **only** from `AWS_REGION` and not at all `~/.aws/config`, **unless**
 the `AWS_SDK_LOAD_CONFIG` environment variable is set. So, we must define our own
 approach to walking AWS config and deciding what to use.
-
 Our chosen approach is:
-
 	"More specific takes precedence over less specific."
-
 1. User-provided configuration is the most explicit.
 2. Environment variables are potentially shared across many invocations and so they have less precedence.
 3. Configuration in `~/.aws/config` is shared across all invocations of a given user and so this has even less precedence.
 4. Configuration retrieved from the EC2 instance metadata service is shared by all invocations on a given machine, and so it has the lowest precedence.
-
 This approach should be used in future updates to this logic.
 */
-func GetRegion(configuredRegion string) (string, error) {
+func GetOrDefaultRegion(logger hclog.Logger, configuredRegion string) string {
```

Comment:
I purposefully moved this out in my changes because other libs using this helper may not have access to an hclog.Logger. Can we keep the error messages and fix the code in Vault to instead look for an error and log if one is found?